snapshot-labs/sx-monorepo
# feat: show proposal's AI summary

### Summary

From the fun call hackathon: 

Show button to fetch and display proposal's AI summary (only for offchain proposal with body length > 500 chars)

This is the default state (show a button to trigger a request to fetch and display the AI summary):

![Screenshot 2024-03-22 at 20 05 17](https://github.com/snapshot-labs/sx-monorepo/assets/495709/9e585d9e-64a6-46fa-9a93-08930320f33b)
 
This is the state with a SUCCESS request:

![Screenshot 2024-03-22 at 20 06 33](https://github.com/snapshot-labs/sx-monorepo/assets/495709/1506911d-7e6b-4177-a284-6912ba3e61d6)

This is the state with an ERROR request (show error message, as well as retry button)

![Screenshot 2024-03-22 at 20 05 31](https://github.com/snapshot-labs/sx-monorepo/assets/495709/d51be4ff-4934-407e-8147-1ec43497f84e)

### How to test

1. Go to am offchain proposal with a body length > 500 chars
2. It should show the "AI Summary" button
3. Click on the button
4. It should show the AI summary after a while for fresh content
5. If you refresh the page and retry, it should show the summary faster, since it's now cached in the backend
6. Go to a proposal with body length < 500 chars
7. It should not show the "Ai Summary" button

### To-Do

- [x] Waiting for backend merge: https://github.com/snapshot-labs/snapshot-sidekick/pull/240
